wifi-subsys/components/protocols/src/icmp_ping.c

```diff
@@ -90,35 +90,66 @@ void check_on_ping_end(esp_ping_handle_t hdl, void *args)
 
 esp_err_t initialize_ping()
 {
+    ESP_LOGI(__func__, "Starting ping setup");
     ip_addr_t target_addr;
     struct addrinfo hint;
     struct addrinfo *res = NULL;
+    char * url;
+    size_t url_len = 0;
     memset(&hint, 0, sizeof(hint));
     memset(&target_addr, 0, sizeof(target_addr));
-    int result = getaddrinfo(DEFAULT_SERVER_TO_PING, NULL, &hint, &res);
-    if((result != 0) || (res == NULL)) {
+    esp_err_t err = nvs_get_string(stringlify(PING),
+                        stringlify(PING_TO),
+                        &url, &url_len);
+    if (err != ESP_OK) {
+        ESP_LOGE(__func__,
+                "error to get url the cause is %s",
+                esp_err_to_name(err));
+
+        return err;
+    }
+        int result = getaddrinfo(&url, NULL, &hint, &res);
+    if ((result != 0) || (res == NULL)) {
         is_connected = FAIL_BIT;
         is_configured = FAIL_BIT;
-        ESP_LOGE(__func__,"DNS lookup failed err=%d res=%p", result, res);
+        ESP_LOGE(__func__, "DNS lookup failed err=%d res=%p", result, res);
         return ESP_FAIL;
     } else {
         is_configured = CONNECTED_BIT;
-        printf("Initialising ping");
+        printf("Initializing ping");
```

Comment:
change this por esp_logi please i know is not your change, but seems we not see this printf when check icmp PR